src/main/java/org/sagebionetworks/bridge/hibernate/HibernateStudyDao.java

```diff
@@ -27,16 +28,21 @@ final void setHibernateHelper(HibernateHelper hibernateHelper) {
     }
 
     @Override
-    public List<Study> getStudies(String appId, boolean includeDeleted) {
+    public PagedResourceList<Study> getStudies(String appId, int offsetBy, int pageSize, boolean includeDeleted) {
         checkNotNull(appId);
         
         Map<String,Object> parameters = ImmutableMap.of("appId", appId);
         String query = "from HibernateStudy as study where appId=:appId";
         if (!includeDeleted) {
             query += " and deleted != 1";
         }
-        return ImmutableList.copyOf(hibernateHelper.queryGet(query, parameters, 
-                null, null, HibernateStudy.class));
+        int total = hibernateHelper.queryCount("select count(*) " + query, parameters);
+
+        List<HibernateStudy> hibStudies = hibernateHelper.queryGet(query, parameters, 
+                offsetBy, pageSize, HibernateStudy.class);
+        List<Study> studies = hibStudies.stream().map(s -> (Study)s).collect(toList());
```

Comment:
I prefer ImmutableList.copyOf(). I think both are probably roughly equally efficient, but copyOf() looks cleaner.

Reply:
I didn't know that would cast to the interface, very nice, changed.